[IMP] l10n_us: allow encoding an intermediary bank

jorenvo opened the pull request:
For international wire transfers it's common to have to specify an intermediary bank [1]. This is needed when the user's bank doesn't have a direct relationship with the recipient's bank.

We add a field for it so that it can be encoded in a structured way. This allows accountants to easily export this data, process it, and upload it to their bank's portal to generate the payments.

task-4252963

[1] https://stripe.com/resources/more/what-are-intermediary-banks-and-how-do-they-work

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/183437.png)](https://mergebot.odoo.com/odoo/odoo/pull/183437)

malb-odoo commented on addons/l10n_us/models/res_bank.py:
```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, fields, api, _
```
nitpicking but alphabetical order :eyes:

malb-odoo commented on addons/l10n_us/models/res_bank.py:
```diff
@@ -0,0 +1,19 @@
+    _inherit = "res.bank"
+
+    intermediary_bank_id = fields.Many2one(
+        "res.bank",
```
while we are at it let's add the comodal_name before res.bank and could be single quotes since it's not something the user can usually see :smile:

malb-odoo commented on addons/l10n_us/models/res_bank.py:
```diff
@@ -0,0 +1,19 @@
+    intermediary_bank_id = fields.Many2one(
+        "res.bank",
+        domain="[('id', '!=', id)]",
+        help="An intermediary bank facilitates international wire transfers between your bank and the beneficiary's bank when they don’t have a direct relationship.",
```
I would still export the pot since we are on a localisation they are not automatically exported. Usually when working on localisation in stable and in master we export pot and make the po since here it's already in english the po is not useful :smile:

jorenvo replied:
[I responded below](https://github.com/odoo/odoo/pull/183437#issuecomment-2560293191).

malb-odoo reviewed (commented):
Just some small nitpicking here and there otherwise it looks good, i see that this is not really a task yet, has this been tested by a po ? :smile:

jorenvo commented:
@malb-odoo thanks for your review! I was told APR agreed and would take ownership of these but that turns out to be wrong and I should ask other product owners instead. I blocked both PRs and will get back to these if they approve.

gawa-odoo commented on addons/l10n_us/__manifest__.py:
```diff
@@ -13,7 +13,8 @@
         'data/res_company_data.xml',
-        'views/res_partner_bank_views.xml'
+        'views/res_partner_bank_views.xml',
+        'views/res_bank_views.xml'
```
You could add a comma here, for a future blame

gawa-odoo reviewed (commented):
Hello
Apparently, this PR should be revived and we would like it to be merged.
Could you rebase and reopen it? (and fix the nitpickings of the malb as you're doing it)

Tell me if I'm mistaken, but I think it's related to accounting/invoicing. If it is, then this should be in the module `l10n_us_account` instead

jorenvo commented:
@gawa-odoo indeed, I moved it to `l10n_us_account` and fixed the issues reported by @malb-odoo. The only outstanding one relates to the translations. [`l10n_us` has used `_()` for a long time without an `i18n` directory](https://github.com/odoo/odoo/blob/9144eb051538d5a92d403e1ad9c635ce39e090ec/addons/l10n_us/models/res_partner_bank.py#L18). I do the same here. I can create the `po{,t}` files, but I'm not sure what the advantage would be. I'd prefer to keep the call to `_()` because that should make it translatable on Transifex when [WAN's work is merged](https://github.com/odoo/odoo/pull/187546) [^1].

[^1]: [The USA has no official languages](https://www.usa.gov/official-language-of-us), but translations for the most commonly spoken languages (e.g. Spanish, Chinese, ...) probably make sense at some point.

gawa-odoo commented:
I agree, thanks. 
@robodoo r+

jorenvo commented:
@gawa-odoo @malb-odoo thanks for reviewing!
